ReactApplication/src/Components/ColorfulGrid.jsx

```diff
@@ -0,0 +1,159 @@
+    const randomColor = colors[randomNumber];
+    return randomColor;
+  }
+
+  //Dynamic Grid
+  const renderGrid = () => {
+    const grid = [];
+    for (let i = 0; i < rows; i++) {
+      for (let j = 0; j < columns; j++) {
+        const randomColor = getRandomColor();
+        grid.push(
+          <GridItem
+          key={`${i}-${j}`}
+          backgroundColor={randomColor}
+        />
+        );
+      }
+    }
+    return grid;
+  };
+
+  const handleColorChange = (newColor) => {
+    setColor(newColor.hex);
+  };
+
+  return (
+    <div className="container">
+      <section className="side-panel">
+        {/* color picker */}
+        <article>
+          <div className="input-label-color cus-style">
+            <label htmlFor="colorInput">Choose Color</label>
+          </div>
+          <SketchPicker
+            color={color}
+            onChange={handleColorChange}
+            className="colorpicker"
+          />
+          <button
+            className="cus-btn cus-style"
+            onClick={() => {
+              setColors([...colors, color]);
+            }}
+          >
+            Add
+          </button>
+        </article>
+
+        {/* row input */}
+        <div className="flex-1">
+          <div className="input-label cus-style">
+            <label htmlFor="rowsInput">R</label>
+          </div>
+          <input
+            type="number"
+            id="rowsInput"
+            min="1"
+            value={rows}
+            className="cus-input"
+            placeholder="Enter no of rows"
+            onChange={(e) => {
+              setRows(e.target.value);
+            }}
+          />
+        </div>
+
+        {/* col input */}
+        <div className="flex-1">
+          <div className="input-label cus-style">
+            <label htmlFor="columnsInput">C </label>
+          </div>
+
+          <input
+            type="number"
+            id="columnsInput"
+            min="1"
+            value={columns}
+            className="cus-input"
+            placeholder="Enter no of Columns"
+            onChange={(e) => setColumns(e.target.value)}
+          />
+        </div>
+
+        {/* color wrapper */}
+        <section className="colors-wrapper">
+          {colors.map((color, index) => {
+            return (
+              <div className="color-picker-div" key={index}>
+                <input
+                  type="color"
+                  value={color}
+                  onChange={(e) =>colorChangeHandler(index, e.target.value)}
+                />
+                {colors.length > 1 && (
+                  <button
+                    className="delete-btn"
+                    onClick={() => onDelete(index)}
+                  >
+                    X
+                  </button>
+                )}
+              </div>
+            );
+          })}
+        </section>
+      </section>
+
+
+        {/* Dynamic Grid */}
+      <main

```

Comment:
format the code properly. get prettier extension. it will help you with formatting or VS code has built in formatters as well.